### PR TITLE
[release-v1.36] Add tolerations for arm64 workloads on GKE

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

By default, GKE schedules workloads only to x86-based nodes by by placing a taint (kubernetes.io/arch=arm64:NoSchedule) on all Arm nodes [1]. This taint prevents Calico components from being scheduled on ARM nodes. This change adds toleration to the workload specification as suggested in [2].

[1] https://cloud.google.com/kubernetes-engine/docs/how-to/prepare-arm-workloads-for-deployment#overview
[2] https://cloud.google.com/kubernetes-engine/docs/how-to/prepare-arm-workloads-for-deployment#multi-arch-schedule-any-arch

Pick https://github.com/tigera/operator/pull/3544 into release v1.36 branch.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
